# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/atk/AccessibleWrap.cpp
# Commit: ed231c100cea
# Full Hash: ed231c100cea9bf7dc2eb1913b89c910ad25c747
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-08-24 09:36:20
# Description:
#   Bug 1786638: Return early if we can't fetch an internal obj when computing ATK relations r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155426
# ==============================================================================

diff -r 079d8864bca3 -r ed231c100cea accessible/atk/AccessibleWrap.cpp
--- a/accessible/atk/AccessibleWrap.cpp	Wed Aug 24 04:29:19 2022 +0000
+++ b/accessible/atk/AccessibleWrap.cpp	Wed Aug 24 04:44:38 2022 +0000
@@ -836,6 +836,10 @@
       ATK_OBJECT_CLASS(parent_class)->ref_relation_set(aAtkObj);
 
   Accessible* acc = GetInternalObj(aAtkObj);
+  if (!acc) {
+    return relation_set;
+  }
+
   if (!StaticPrefs::accessibility_cache_enabled_AtStartup() &&
       acc->IsRemote()) {
     RemoteAccessible* proxy = acc->AsRemote();
